docs/unsettled-fees.md

```diff
@@ -2,76 +2,87 @@
+#### Path A: Order Cancellation by User
 
-The settlement logic is as follows:
+If a user cancels their own order, the associated `UserUnsettledFee` is settled immediately and completely. The logic is as follows:
 
-- The system checks how much of the order was filled versus unfilled.
-- Unsettled fees are split proportionally:
-  - **Unfilled portion**: Fees are returned directly to the user.
-  - **Filled portion**: Fees are paid directly to the protocol's treasury.
+- The fee for the **unfilled portion** of the order is returned directly to the user as a `Coin`.
+- The fee for the **filled portion** is moved into the `protocol_unsettled_fees` bag for later collection.
+- The original `UserUnsettledFee` object is destroyed, granting the user an immediate gas storage rebate.
 
-### 2. Protocol Collection Settlement
+#### Path B: Order Completion or Protocol-Side Collection
 
-The permissionless `settle_protocol_fee_and_record` function serves as a mechanism to settle fees for orders that are no longer live but haven't been settled by the user. Its primary use cases are:
+When an order is completed (e.g., fully filled) or cancelled externally, the fees can be settled permissionlessly by anyone through a batch process.
 
-- **Fully Filled Orders**: To collect fees after an order has been completely filled.
-- **Externally Cancelled Orders**: To collect fees for orders that were cancelled outside of the Deeptrade protocol's fee-settling functions. In this case, the entire unsettled fee is paid to the protocol.
+1.  **Settle Filled Order Fees**: Anyone can call `settle_filled_order_fee_and_record` for any finalized order. This function drains the entire fee from the `UserUnsettledFee` object and sends it directly to the protocol treasury. It intentionally leaves the now-empty `UserUnsettledFee` object in the bag.
 
-This ensures that all fees from finalized orders are eventually collected, even if the user does not manually settle them.
+2.  **Settle Protocol Fees**: Anyone can call `settle_protocol_fee_and_record`. This function drains the aggregated balances from the `protocol_unsettled_fees` bag and sends them to the treasury. It also leaves empty `Balance` objects in the bag.
+
+### 3. Storage Rebate Claims
+
+The permissionless settlement functions (Path B) intentionally leave empty objects in the bags. This is a crucial design choice that separates fee collection from storage rebate claims.
+
+- The original owner of the `FeesManager` (or a protocol admin) must call one of the `claim_*_storage_rebate` functions to destroy these empty objects and reclaim the initial storage deposit.
+- This two-step process separates permissionless fee collection from permissioned rebate claims. It allows anyone to help settle protocol fees, while ensuring that only the rightful owner of the `FeesManager` can claim the storage rebate for the object they initially paid to create.
 
```

Comment:
tbh mb we should say a bit why did we introduce FeeManager?
or even mb create a separate doc? idk